pallets/asset/src/lib.rs

```diff
@@ -1362,7 +1380,8 @@ impl<T: Config> Module<T> {
         );
 
         // Next global key.
-        let key = AssetMetadataNextGlobalKey::try_mutate(try_next_pre::<T, _>)?;
+        let key = Self::update_current_asset_metadata_global_key()?;
+        AssetMetadataNextGlobalKey::try_mutate(try_next_pre::<T, _>)?;
```

Comment:
Just set the old storage value to `key`.  The mutate operation has an extra read that we don't need.